adapters/httpapi_compact.c

```diff
@@ -1354,19 +1367,32 @@ HTTPAPI_RESULT HTTPAPI_ExecuteRequest(HTTP_HANDLE handle, HTTPAPI_REQUEST_TYPE r
     HTTP_HEADERS_HANDLE responseHeadersHandle, BUFFER_HANDLE responseContent)
 {
     return HTTPAPI_ExecuteRequest_Internal(handle, requestType, relativePath,
-        httpHeadersHandle, content, contentLength, statusCode, responseHeadersHandle, responseContent, NULL, NULL);
+        httpHeadersHandle, content, contentLength, statusCode, NULL, responseHeadersHandle, responseContent, NULL, NULL);
 }
 
-/*Codes_SRS_HTTPAPI_COMPACT_21_021: [ The HTTPAPI_ExecuteRequest shall execute the http communtication with the provided host, sending a request and reciving the response. ]*/
+/*Codes_SRS_HTTPAPI_COMPACT_21_021: [ The HTTPAPI_ExecuteRequest shall execute the http communication with the provided host, sending a request and receiving the response. ]*/
+/*Codes_SRS_HTTPAPI_COMPACT_21_050: [ If there is a content in the response, the HTTPAPI_ExecuteRequest shall copy it in the responseContent buffer. ]*/
+//Note: This function assumes that "Host:" and "Content-Length:" headers are setup
+//      by the caller of HTTPAPI_ExecuteRequest() (which is true for httptransport.c).
+HTTPAPI_RESULT HTTPAPI_ExecuteRequest_With_Reason_Phrase(HTTP_HANDLE handle, HTTPAPI_REQUEST_TYPE requestType, const char* relativePath,
+    HTTP_HEADERS_HANDLE httpHeadersHandle, const unsigned char* content,
+    size_t contentLength, unsigned int* statusCode, char* reasonPhrase,
```

Comment:
need add one more parameter telling length of the buffer, otherwise there could be buffer over run